prover/src/lib.rs

```diff
@@ -1,4 +1,13 @@
-#![cfg_attr(not(feature = "std"), no_std)]
+#![no_std]
+
+#[cfg_attr(
+    all(feature = "std", feature = "metal", target_arch = "aarch64", target_os = "macos"),
+    macro_use
+)]
```

Comment:
Ah if that's the case, then yeah it can be removed. The issue I was trying to solve is that the macros are only used when all of those conditions are true, so I didn't want to have some combination of features slip by that resulted in an unused warning, but no sense being redundant and making it more complicated than it needs to be.